test/e2e/daemon_set.go

```diff
@@ -510,19 +510,23 @@ func setDaemonSetNodeLabels(c clientset.Interface, nodeName string, labels map[s
 	return newNode, nil
 }
 
-func checkDaemonPodOnNodes(f *framework.Framework, selector map[string]string, nodeNames []string) func() (bool, error) {
+func checkDaemonPodOnNodes(f *framework.Framework, ds *extensions.DaemonSet, nodeNames []string) func() (bool, error) {
 	return func() (bool, error) {
-		selector := labels.Set(selector).AsSelector()
-		options := metav1.ListOptions{LabelSelector: selector.String()}
-		podList, err := f.ClientSet.Core().Pods(f.Namespace.Name).List(options)
+		podList, err := f.ClientSet.Core().Pods(f.Namespace.Name).List(metav1.ListOptions{})
 		if err != nil {
 			return false, nil
 		}
 		pods := podList.Items
 
 		nodesToPodCount := make(map[string]int)
 		for _, pod := range pods {
-			if controller.IsPodActive(&pod) {
+			if controllerRef := controller.GetControllerOf(&pod); controllerRef == nil || controllerRef.UID != ds.UID {
+				continue
+			}
+			if pod.DeletionTimestamp != nil {
+				continue
+			}
+			if v1.IsPodAvailable(&pod, ds.Spec.MinReadySeconds, metav1.Now()) {
 				nodesToPodCount[pod.Spec.NodeName] += 1
 			}
 		}
```

Comment:
Also in L538 where false is returned because more than one pod is running in a node, add an additional log

"Node %q is running more than one daemon pod..."